packages/server/src/jsRunner/vm/index.ts

```diff
@@ -127,6 +131,68 @@ export class IsolatedVM implements VM {
 
   withContext(context: Record<string, any>) {
     this.addToContext(context)
+
+    return this
+  }
+
+  withParsingBson(data: any) {
+    this.addToContext({
+      bsonData: bson.BSON.serialize({ data }),
+    })
+
+    // If we need to parse bson, we follow the next steps:
+    // 1. Serialise the data from potential BSON to buffer before passing it to the isolate
+    // 2. Deserialise the data within the isolate, to get the original data
+    // 3. Process script
+    // 4. Stringify the result in order to convert the result from BSON to json
```

Comment:
How come we don't repeat the process of serializing BSON inside the isolate and deserializing it ouside, to make the process truly transparent?

Reply:
I did it this way because I don't see any specific benefit, and we don't need a post-process conversion, as the output will be already a valid JSON. If needed, we can extend the logic to handle it